crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -295,6 +298,39 @@ impl PublishFlakeRef {
             PublishFlakeRef::Https(https_ref) => FlakeRef::GitHttps(https_ref),
         }
     }
+
+    #[allow(unused)]
+    fn from_github_ref(
+        GitServiceRef {
+            owner,
+            repo,
+            attributes,
+            ..
+        }: GitServiceRef<service::Github>,
+    ) -> Result<Self, ConvertFlakeRefError> {
+        let host = attributes.host.unwrap_or("github.com".to_string());
+
+        let url_str = format!("https://{host}/{owner}/{repo}");
+        let url = WrappedUrl::from_str(&url_str)
+            .map_err(|e| ConvertFlakeRefError::InvalidResultUrl(url_str, e))?;
+
+        let only_rev = attributes.rev.is_some() && attributes.reference.is_none();
+
+        let git_attributes = GitAttributes {
+            rev: attributes.rev,
+            reference: attributes.reference,
+            dir: attributes.dir,
+            all_refs: only_rev.then_some(true),
+            ..Default::default()
+        };
+
+        let git_ref = GitRef {
+            url,
+            attributes: git_attributes,
+        };
```

Comment:
(publishing) users will more likely have ssh configured rather than (often insecure) token access.
In cases where https token access is granted by another tool (e.g. `gh`) or within github actions we can prefer https urls so not to require an ssh setup to be present.